ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeMessagePool.java

```diff
@@ -92,18 +92,21 @@ private synchronized void doAdd(final ValidateableSyncCommitteeMessage message)
         message.getSubcommitteeAssignments().orElseThrow();
     final Map<BlockRootAndCommitteeIndex, ContributionData> blockRootAndCommitteeIndexToMessages =
         committeeContributionData.computeIfAbsent(message.getSlot(), __ -> new HashMap<>());
-    assignments
-        .getAssignedSubcommittees()
-        .forEach(
-            subcommitteeIndex ->
-                blockRootAndCommitteeIndexToMessages
-                    .computeIfAbsent(
-                        new BlockRootAndCommitteeIndex(
-                            message.getBeaconBlockRoot(), subcommitteeIndex),
-                        __ -> new ContributionData())
-                    .add(
-                        assignments.getParticipationBitIndices(subcommitteeIndex),
-                        message.getMessage().getSignature()));
+    IntSet applicableSubnets;
```

Comment:
This is where the usage of `final` makes more sense :)

Reply:
Ashes on m head, thank you!